src/client.rs

```diff
@@ -173,7 +157,25 @@ where T: AsyncRead + AsyncWrite {
     type Error = ::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        self.inner.poll()
+        let (io, _) = try_ready!(self.inner.poll());
+
+        debug!("client connection bound");
+
+        // Create the codec
+        let mut codec = Codec::new(io);
+
+        // Create the initial SETTINGS frame
+        let settings = self.config.settings.clone();
+
+        // Send initial settings frame
+        match codec.start_send(settings.clone().into()) {
```

Comment:
Since you are here, could you switch this to use [`Codec::buffer`](https://github.com/carllerche/h2/blob/9448a194081e943e5768bc874d39aa7fe842196a/src/codec/mod.rs#L129) instead.